### PR TITLE
k8s VNEXT agents cluster template

### PR DESCRIPTION
**Pull request by EwanNoble**

### CODE REVIEW

File: cluster.template.json [hunk 1]
```diff
@@ -1 +1,178 @@
...
+    "clusterName": "[concat(variables('resourceNamePrefix'), '-aks')]",
+    "vnetName": "[concat(variables('resourceNamePrefix'), '-vnet')]",
+    "subnetName": "agent-aks",
+    "agentPools": [
+      // {
+      //   "name": "Linux",
+      //   "os": "Linux",
+      //   "nodeCount": "[parameters('linuxNodeCount')]",
+      //   "vmSize": "Standard_DS2_v2"
+      // },
```
Comment (chelnak) [inline cluster.template.json]: ?

File: cluster.template.json [hunk 2]
```diff
@@ -1 +1,178 @@
...
+            }
+          }
+        ],
+        "addressSpace": {
+          "addressPrefixes": [
+            "10.0.0.0/8"
+          ]
+        }
+      }
+    },
```
Comment (chelnak) [inline cluster.template.json]: Should we reduce this network space?

File: cluster.template.json (hunk 1 above)
Comment (EwanNoble) [reply cluster.template.json]: Saving money for now

File: cluster.template.json [hunk 3]
```diff
@@ -72,13 +72,13 @@
...
             "properties": {
-              "addressPrefix": "10.240.0.0/16"
+              "addressPrefix": "10.1.0.0/16"
             }
           }
         ],
         "addressSpace": {
           "addressPrefixes": [
-            "10.0.0.0/8"
+            "10.0.0.0/15"
```
Comment (chelnak) [inline cluster.template.json]:
```suggestion
            "10.0.0.0/18"
```

**Pull request merged**